File: src/main/java/me/kaotich00/fwauctionhouse/storage/Storage.java
package me.kaotich00.fwauctionhouse.storage;

import me.kaotich00.fwauctionhouse.FwAuctionHouse;

public class Storage {

    private final FwAuctionHouse plugin;
    private final StorageMethod storageMethod;

    public Storage(FwAuctionHouse plugin, StorageMethod storageMethod) {
        this.plugin = plugin;
        this.storageMethod = storageMethod;
    }

    public StorageMethod getStorageMethod() {
        return this.storageMethod;
    }

    public void init() {
        try {
            this.storageMethod.init();
        } catch (Exception e) {
            this.plugin.getLogger().severe("Failed to init storage implementation");
            e.printStackTrace();
        }
    }

    public void shutdown() {
        try {
            this.storageMethod.shutdown();
        } catch (Exception e) {
            this.plugin.getLogger().severe("Failed to shutdown storage implementation");
            e.printStackTrace();
        }
    }

}

File: src/main/java/me/kaotich00/fwauctionhouse/storage/Storage.kt
package me.kaotich00.fwauctionhouse.storage

import me.kaotich00.fwauctionhouse.FwAuctionHouse

class Storage(private val plugin: FwAuctionHouse, val storageMethod: StorageMethod) {
    fun init() {
        try {
            storageMethod.init()
        } catch (e: Exception) {
            plugin.logger.severe("Failed to init storage implementation")
            e.printStackTrace()
        }
    }

    fun shutdown() {
        try {
            storageMethod.shutdown()
        } catch (e: Exception) {
            plugin.logger.severe("Failed to shutdown storage implementation")
            e.printStackTrace()
        }
    }
}